include/zephyr/bluetooth/hci.h

```diff
@@ -97,6 +97,15 @@ int bt_hci_get_conn_handle(const struct bt_conn *conn, uint16_t *conn_handle);
  */
 int bt_hci_get_adv_handle(const struct bt_le_ext_adv *adv, uint8_t *adv_handle);
 
+/** @brief Get periodic advertising sync handle for an advertising set.
+ *
+ * @param sync Periodic advertising sync set.
+ * @param sync_handle Place to store the periodic advertising sync handle.
+ *
+ * @return 0 on success or negative error value on failure.
+ */
+int bt_hci_get_sync_handle(const struct bt_le_per_adv_sync *sync, uint16_t *sync_handle);
```

Comment:
I think it'd be appropriate to have "adv" in the function name, since you have it in the first parameter type as well. It also makes it a bit clearer what "sync" we're talking about (as "sync" exists even on the BR/EDR side for SCO links). The name would then be `bt_hci_get_adv_sync_handle()`